cirq/ops/pauli_string_phasor.py

```diff
@@ -0,0 +1,180 @@
+# Copyright 2018 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Dict, Iterable, Union, Callable
+
+import sympy
+
+from cirq import value, protocols
+from cirq.ops import (raw_types, common_gates, pauli_string as ps, pauli_gates,
+                      op_tree, pauli_string_raw_types)
+
+
+@value.value_equality(approximate=True)
+class PauliStringPhasor(pauli_string_raw_types.PauliStringGateOperation):
+    """An operation that phases a Pauli string."""
+
+    def __init__(self,
+                 pauli_string: ps.PauliString,
+                 *,
+                 exponent_neg: Union[int, float, sympy.Basic] = 1,
+                 exponent_pos: Union[int, float, sympy.Basic] = 0) -> None:
+        """Initializes the operation.
+
+        At most one angle argument may be specified. If more are specified,
```

Comment:
I don't see this in the code

Reply:
Cancelled merged cause this wasn't addressed.

Reply:
Fixed.